services/notify/src/api.js

```diff
@@ -86,6 +114,110 @@ builder.declare({
     channel: input.channel,
     user: input.user,
   });
-  await this.notifier.irc(input);
+
+  let notificationAddress = {
+    notificationType: input.user ? "irc-user" : "irc-channel",
+    notificationAddress: input.user ? input.user : input.channel,
+  };
+  // Ensure that the address is not in the blacklist
+  let response = await this.BlacklistedNotification.load(notificationAddress, true);
+  if(!response) {
+    await this.notifier.irc(input);
+    res.sendStatus(200);
+  } else {
+    return res.reportError('BlacklistedAddress', `${input.channel || input.user} is blacklisted`, {});
+  }
+});
+
+builder.declare({
+  method: 'post',
+  route: '/blacklist/add',
+  name: 'addBlacklistAddress',
+  scopes: 'notify:manage-blacklist:<notificationType>/<notificationAddress>',
+  input: 'notification-address.yml',
+  title: 'Blacklist Given Address',
+  description: [
+    'Add the given address to the notification blacklist. The address',
+    'can be of either of the three supported address type namely pulse, email',
+    'or IRC(user or channel). Addresses in the blacklist will be ignored',
+    'by the notification service.',
+  ].join('\n'),
+}, async function(req, res) {
+  // The address to blacklist
+  let address = Object.assign({}, {
```

Comment:
I think the `Object.assign` isn't required here?  Just `= { notificationType: .., notificationAddress: ..}` should create a new object for you.

Reply:
haha you're right, I don't even remember why I used it there